ElectronClient/gui/KeymapConfig/ShortcutRecorder.tsx

```diff
@@ -0,0 +1,65 @@
+import * as React from 'react';
+import { useState, KeyboardEvent } from 'react';
+
+import styles_ from './styles';
+import KeymapService from '../../lib/services/KeymapService';
+import { KeymapConfigScreenProps } from './KeymapConfigScreen';
+
+const { _ } = require('lib/locale.js');
+const keymapService = KeymapService.instance();
+
+export interface ShortcutRecorderProps extends KeymapConfigScreenProps {
+	setAccelerator: (accelerator: string) => void,
+	resetAccelerator: () => void,
+	toggleEditing: () => void,
+}
+
+export const ShortcutRecorder = ({ setAccelerator, resetAccelerator, toggleEditing, theme }: ShortcutRecorderProps) => {
```

Comment:
The child component logic should be decoupled from that of the parent. ShortcutRecorder should only take three event handlers: onSave, onCancel, onReset then on the parent component you handle the business logic to set keymap.

In fact I don’t see anything in this component to cancel input but that should be present

Reply:
I've updated the logic to use handlers instead. Thanks for pointing that out.